pkg/coverage/jacoco.go

```
@@ -0,0 +1,113 @@
+	XMLName     xml.Name `xml:"report"`
+	Text        string   `xml:",chardata"`
+	Name        string   `xml:"name,attr"`
+	Sessioninfo []struct {
+		Text  string `xml:",chardata"`
+		ID    string `xml:"id,attr"`
+		Start string `xml:"start,attr"`
+		Dump  string `xml:"dump,attr"`
+	} `xml:"sessioninfo"`
+	Packages []struct {
+		Text    string `xml:",chardata"`
+		Name    string `xml:"name,attr"`
+		Classes []struct {
+			Text           string `xml:",chardata"`
+			Name           string `xml:"name,attr"`
+			Sourcefilename string `xml:"sourcefilename,attr"`
+			Method         []struct {
+				Text    string          `xml:",chardata"`
+				Name    string          `xml:"name,attr"`
+				Desc    string          `xml:"desc,attr"`
+				Line    string          `xml:"line,attr"`
+				Counter []jacocoCounter `xml:"counter"`
+			} `xml:"method"`
+			Counter []jacocoCounter `xml:"counter"`
+		} `xml:"class"`
+		Sourcefiles []struct {
+			Text string `xml:",chardata"`
+			Name string `xml:"name,attr"`
+			Line []struct {
+				Text string `xml:",chardata"`
+				Nr   string `xml:"nr,attr"`
+				Mi   string `xml:"mi,attr"`
+				Ci   string `xml:"ci,attr"`
+				Mb   string `xml:"mb,attr"`
+				Cb   string `xml:"cb,attr"`
+			} `xml:"line"`
+			Counter []jacocoCounter `xml:"counter"`
+		} `xml:"sourcefile"`
+		Counter []jacocoCounter `xml:"counter"`
+	} `xml:"package"`
+	Counter []jacocoCounter `xml:"counter"`
+}
+
+func countJacoco(c *Coverage, counter *jacocoCounter) {
+	switch counter.Type {
+	case "LINE":
+		c.LinesFound += counter.Covered + counter.Missed
+		c.LinesHit += counter.Covered
+	case "BRANCH":
+		c.BranchesFound += counter.Covered + counter.Missed
+		c.BranchesHit += counter.Covered
+	case "METHOD":
+		c.FunctionsFound += counter.Covered + counter.Missed
+		c.FunctionsHit += counter.Covered
+	}
+}
+
+func ParseJacocoXML(reportPath string) *CoverageSummary {
+	summary := &CoverageSummary{
+		Total: &Coverage{},
+	}
+
+	data, _ := os.ReadFile(reportPath)
```

Shouldn't we return this error rather than ignoring it?